bitcoin/src/psbt/mod.rs

```diff
@@ -44,15 +44,55 @@ use self::map::Map;
 /// Partially signed transaction, commonly referred to as a PSBT.
 pub type Psbt = PartiallySignedTransaction;
 
+/// Future version of PSBT this library can't parse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct FutureVersionError(u32);
+
+impl fmt::Display for FutureVersionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("Unknown future PSBT version ")?;
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+#[cfg(feature = "std")]
+impl ::std::error::Error for FutureVersionError {}
+
+/// Partially signed transaction version as defined in BIP-174 and BIP-370.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+#[non_exhaustive]
+pub enum Version {
+    /// PSBT version 0, defined in BIP-174
+    PsbtV0 = 0,
+    /// PSBT version 2, defined in BIP-370
+    PsbtV2 = 2,
+}
+
+impl Version {
+    fn to_standard(self) -> u32 {
+        self as u32
+    }
+
+    fn from_standard(version: u32) -> Result<Version, FutureVersionError> {
```

Comment:
The naming `to_standard` and `from_standard` doesn't make much sense since this is not a consensus object. Neither `to_consensus`/`from_consensus`.

`to_raw` or `to_u32` or `to_num` would make more sense, IDK which is best.

Reply:
Sure it's not consensus per se but it is `to_standard` in that there is a standard spec that gets referenced at the very top of the file. Are you sure that name implies consensus and is therefore a problem?

Reply:
`standard` in the codebase has specific meaning: it's what the Bitcoin nodes consider standard for a transaction. E.g. it is valid to create a transaction output with 0 amount (can be mined in a block) but it's not standard (nodes won't relay the transaction). Some  types have `to_standard` to convey "the value you get from this will definitely be accepted by bitcoin nodes" and `from_standard` "this implements the same standardness policy as current Bitcoin nodes so valid values are still rejected if not standard"

None of this obviously applies to PSBT because those are not transferred over the P2P network.

Reply:
What's the difference between `to_consensus` and `to_standard` then?

Reply:
IIUC "consensus" is a subset of "standard". So if we guarantee one or the other we use the strictest i.e., if we guarantee "standard" we use `to_standard` and not `to_consensus` even though it is also "consensus". Please correct me if I'm wrong @Kixunil because this is my current understanding.

Reply:
Yes, `to_standard` implies `to_consensus` but not vice-versa. If you're validating blocks you need `from_consensus` (theoretically; in reality you use `consensus_decode`) but if you're validating whether the transaction will be relayed you need `from_standard`.

Reply:
Neat! `to_raw` it is.